tests/integration/test_templates.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import logging
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import get_unit_address, is_alertmanager_up
+from pytest_operator.plugin import OpsTest
+from werkzeug.wrappers import Request, Response
+
+from alertmanager_client import Alertmanager
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+app_name = METADATA["name"]
+resources = {"alertmanager-image": METADATA["resources"]["alertmanager-image"]["upstream-source"]}
+
+
+def request_handler(request: Request):
+    response = Response("OK", status=200, content_type="text/plain")
+    logger.info("Got Request Data : %s", json.loads(request.data.decode("utf-8")))
+    return response
+
+
+@pytest.mark.abort_on_fail
+async def test_receiver_gets_alert(ops_test: OpsTest, charm_under_test, httpserver):
+
+    # deploy charm from local source folder
+    await ops_test.model.deploy(charm_under_test, resources=resources, application_name=app_name)
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=1000)
+    assert ops_test.model.applications[app_name].units[0].workload_status == "active"
+    assert await is_alertmanager_up(ops_test, app_name)
+
+    # define the alertmanager configuration
+    receiver_name = "fake-receiver"
+    aconfig = {
+        "global": {"http_config": {"tls_config": {"insecure_skip_verify": True}}},
+        "route": {
+            "group_by": ["alertname"],
+            "group_wait": "3s",
+            "group_interval": "5m",
+            "repeat_interval": "1h",
+            "receiver": receiver_name,
+        },
+        "receivers": [
+            {
+                "name": receiver_name,
+                "slack_configs": [
+                    {
+                        "api_url": httpserver.url_for("/"),
+                        "channel": "test",
+                        "text": r"https://localhost/alerts/{{ .GroupLabels.alertname }}",
+                    }
+                ],
+            }
+        ],
+    }
+
+    # use a template to define the slack callback id
+    atemplate = r'{{ define "slack.default.callbackid" }}2{{ end }}'
+    # set alertmanager configuration and template file
+    await ops_test.model.applications[app_name].set_config(
```

Comment:
Add assertion that the template is not in effect before this call? Otherwise you could pass `config=` as part of `deploy`.

Reply:
Not sure what you mean. Clearly no template has been set before this point !

Reply:
In that case I would
```python
await ops_test.model.deploy(
    charm_under_test, 
    resources=resources, 
    application_name=app_name,
    config=your_config,
)
```

Reply:
Leaving as is for now but will gladly change it if there is stronger reason than a preference.